pkg/mock/handlers/handlers.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/mock/dynamic"
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/mock/static"
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/server"
+)
+
+const (
+	// shortestRouteLength represents smallest metadata path without prefix + "/" ex: "mac/"
+	shortestRouteLength = 4
+	versionsPath        = "/"
+	latestPath          = "/latest"
+)
+
+var (
+	routeLookupTable = make(map[string][]string)
```

Comment:
I still think the catch all handler can be simplified and the optimization of storing the subpaths might be an overkill here. 

A simple example here - https://play.golang.org/p/PBIg_H8gtuA. 
We can talk more,  I might be missing some context.